src/frontend/src/optimizer/plan_node/stream_materialize.rs

```diff
@@ -108,14 +110,32 @@ impl StreamMaterialize {
         let pk_indices = &base.logical_pk;
 
         let mut col_names = HashSet::new();
-        for name in &out_names {
-            if !col_names.insert(name.clone()) {
-                return Err(
-                    InternalError(format!("column {} specified more than once", name)).into(),
-                );
+        let mut out_name_iter = if let Some(user_assign_names) = user_assign_names {
+            for name in &user_assign_names {
+                if !col_names.insert(name.clone()) {
+                    return Err(
+                        InternalError(format!("column {} specified more than once", name)).into(),
+                    );
+                }
             }
-        }
-        let mut out_name_iter = out_names.into_iter();
+            // Len of user_assign_names(User specified column names) should equal to output columns.
+            if user_assign_names.len() != out_names.len() {
+                return Err(InternalError(
```

Comment:
Should we use other error types instead of `InternalError` here? cc @st1page @xiangjinwu

Reply:
`InvalidInputSyntax`?

Reply:
Oh it's a semantics error, not a syntax error 🤔